### DISCUSSION TITLE
Is there an easier way to get the status of all tasks running on the scheduler?

### DISCUSSION THREAD

**Discussion by dask-user** [dask/dask#9745] [Q&A]
I can currently see the status of all tasks on the scheduler like so:

```
client.run_on_scheduler(lambda dask_scheduler: dask_scheduler.get_task_status(dask_scheduler.tasks.keys()))
```

However, is there a more concise built in way to get this info?

**Comment by fjetter** [comment]
I wouldn't necessarily classify this information as private but it is very low level and users typically do not interact with dask on this level. 
What are you trying to achieve? Maybe we have something else that fits your need.

**Reply by dask-user** [reply]
For a couple reasons:

1. Extending the monitoring capabilities of the UI. In the UI it can be hard to determine which tasks are in which state since they cannot be searched. In DAGs with predictable and well known structures this can be desirable, for example to estimate progress of the run. Or in cases where a run is hung up, to find which tasks are stuck in a processing state.

2. In cases where information relevant to the dag run is held externally to dask. E.g. for custom data passing optimized for specific workflows that perform better than the general built in solution, being able to identify all tasks which are in a "released" state to periodically perform a cleanup of the custom system, during the run.
    - For the last example, something like state transition hooks might be interesting, but not something I would ever consider necessary, and would honestly probably lead to more problems than it would solve.

**Reply by fjetter** [reply]
> Extending the monitoring capabilities of the UI

I assume you are talking about an UI you are building, not our bokeh dashboard. The only thing I can point you to are various progressbar implementations, see https://docs.dask.org/en/stable/diagnostics-distributed.html#progress-bar / https://github.com/dask/distributed/blob/main/distributed/diagnostics/progress.py

> In cases where information relevant to the dag run is held externally to dask. E.g. for custom data passing optimized for specific workflows that perform better than the general built in solution, being able to identify all tasks which are in a "released" state to periodically perform a cleanup of the custom system, during the run.

We don't offer anything to sync state. transition hooks are the only other thing you could try instead of `Client.run_on_scheduler` but generally speaking I do not recommend to sync state of individual tasks. This can be pretty high volume and slow down the scheduler.
